type_checking/types.py

```diff
@@ -184,6 +184,20 @@ def __hash__(self) -> int:
         return hash(self.text_repr())
 
 
+class Void(Type):

```

Comment:
I don't think we need the void type, because we have native untyped pointers in cont